tasks/ci.rake

```diff
@@ -243,13 +256,37 @@ end
 def beaker_suite(type)
   beaker(:init, '--hosts', ENV['HOSTS'], '--options-file', "config/#{String(type)}/options.rb")
   beaker(:provision)
-  beaker(:exec, 'pre-suite', '--preserve-state', '--pre-suite', pre_suites(type))
-  beaker(:exec, 'pre-suite', '--preserve-state')
-  beaker(:exec, ENV['TESTS'])
-  beaker(:exec, 'post-suite')
+  begin
+    beaker(:exec, 'pre-suite', '--preserve-state', '--pre-suite', pre_suites(type))
+    beaker(:exec, 'pre-suite', '--preserve-state')
+    beaker(:exec, ENV['TESTS'])
+    beaker(:exec, 'post-suite')
+  ensure
+    preserve_hosts = ENV['OPTIONS'].include?('--preserve-hosts=always') if ENV['OPTIONS']
+    beaker(:destroy) unless preserve_hosts
+  end
+end
+
+def beaker_suite_retry(type)
+  beaker(:init, '--hosts', ENV['HOSTS'], '--options-file', "config/#{String(type)}/options.rb")
+  beaker(:provision)
+  begin
+    beaker(:exec, 'pre-suite', '--preserve-state', '--pre-suite', pre_suites(type))
+    beaker(:exec, 'pre-suite', '--preserve-state')
+
+    beaker(:exec, ENV['TESTS'], '--save-results', '/tmp/results.json')
+  rescue RuntimeError
+    tests_to_rerun = get_tests_from('/tmp/results.json')
```

Comment:
Another thing I was thinking about was to output the fact that we were retrying, and which spec files were included in the retry?

Reply:
Done.

Reply:
Done